stingray/crosscorrelation.py

```diff
@@ -193,9 +194,15 @@ def _make_corr(self, lc1, lc2):
         # Calculates cross-correlation of two lightcurves
         self.corr = \
             signal.correlate(lc1_counts, lc2_counts, self.mode)
+        # Obtains correlation lags 
+        # signal.correlation_lags() method uses SciPy versions >= 1.6.1
+        x_lags = \
```

Comment:
Please note that we used the internal method `cal_timeshift` here, that remains buggy despite your fix. It would be better to move this operation to that method, so that it's solved everywhere.

Reply:
The number you can input for both light curve sizes is `self.n` in that case

Reply:
Thanks, @matteobachetti, I have shifted the operations to the internal method `cal_timeshift` as you suggested. Let me know what would be the next step.

Reply:
Understanding why the test fails :).